File: 09 Valid Perfect Square/Valid Perfect Square.cpp
#define ll long long
class Solution
{
public:
    bool isPerfectSquare(int num)
    {
        ll low = 1, high = 100000;
        while (low <= high)
        {
            ll mid = low + (high - low) / 2;
            ll sq = mid * mid;
            if (sq == num)
                return true;
            else if (sq > num)
                high = mid - 1;
            else
                low = mid + 1;
        }
        return false;
    }
};

File: 09 Valid Perfect Square/Valid Perfect Square.js
/**
 * @param {number} num
 * @return {boolean}
 */
var isPerfectSquare = function (num) {
    let low = 1, high = 100000;
    while (low <= high) {
        let mid = low + Math.floor((high - low) / 2);
        let sq = mid * mid;
        if (sq == num)
            return true;
        else if (sq > num)
            high = mid - 1;
        else
            low = mid + 1;
    }
    return false;
};